ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/ClientImpl.java

```diff
@@ -455,6 +456,33 @@ public void close() {
     );
   }
 
+  @Override
+  public HttpRequest buildRequest(final String method, final String path) {
+    return new HttpRequestImpl(method, path, this);
+  }
+
+  CompletableFuture<HttpResponse> send(final HttpRequest request) {
+    final CompletableFuture<HttpResponse> cf = new CompletableFuture<>();
+
+    Map<String, Object> payload = request.payload();
+    // include properties
+    payload.put("properties", request.properties());
```

Comment:
The /ksql and /query endpoints use the parameter `streamsProperties` instead of `properties`. We can either add add some logic here to use the right parameter or remove this and force the user to add the payload themselves. I'd be okay with either approach.

Reply:
ah ok, let me remove this function then. thanks!

Reply:
on second thought, let me try adding a new function that allows users to set the right parameter. I do think having a way to include properties piece-wise is quite handy, so might be worth keeping it.